src/core/service.c

```diff
@@ -480,15 +481,15 @@ static int on_fd_store_io(sd_event_source *e, int fd, uint32_t revents, void *us
         return 0;
 }
 
-static int service_add_fd_store(Service *s, int fd, const char *name, bool do_poll) {
+static int service_add_fd_store(Service *s, int fd_in, const char *name, bool do_poll) {
+        _cleanup_(service_fd_store_unlinkp) ServiceFDStore *fs = NULL;
+        _cleanup_(asynchronous_closep) int fd = ASSERT_FD(fd_in);
         struct stat st;
-        ServiceFDStore *fs;
         int r;
 
-        /* fd is always consumed if we return >= 0 */
+        /* fd is always consumed even if failed. */
```

Comment:
"if failed" would mean that the fd is failed.
"if the function fails" or "if we fail" or "if this fails".